application/common/models/User.php

```diff
@@ -210,7 +210,7 @@ private function validateExpiration(): Closure
     {
         return function ($attributeName) {
             if ($this->currentPassword !== null) {
-                $gracePeriodEnds = strtotime($this->currentPassword->grace_period_ends_utc);
+                $gracePeriodEnds = strtotime("{$this->currentPassword->grace_period_ends_on} 23:59:59");
```

Comment:
If we make `... 23:59:59` instead be `... 23:59:59 UTC`, it might further reduce timezone offset errors.